src/program/content/trakt.py

```diff
@@ -73,18 +76,33 @@ def missing(self):
             logger.log("TRAKT", "Trending fetching is disabled.")
         if not self.settings.fetch_popular:
             logger.log("TRAKT", "Popular fetching is disabled.")
+        if not self.settings.fetch_most_watched:
+            logger.log("TRAKT", "Most watched fetching is disabled.")
 
     def run(self):
         """Fetch media from Trakt and yield Movie, Show, or MediaItem instances."""
+        global last_update
 
         watchlist_ids = self._get_watchlist(self.settings.watchlist) if self.settings.watchlist else []
         collection_ids = self._get_collection(self.settings.collection) if self.settings.collection else []
         user_list_ids = self._get_list(self.settings.user_lists) if self.settings.user_lists else []
-        trending_ids = self._get_trending_items() if self.settings.fetch_trending else []
-        popular_ids = self._get_popular_items() if self.settings.fetch_popular else []
+
+        # Check if it's the first run or if a day has passed since the last update
+        current_time = datetime.now()
+        if last_update is None or (current_time - last_update) > timedelta(days=1):
```

Comment:
I was thinking to just add this to those functions instead, but this works too 😁